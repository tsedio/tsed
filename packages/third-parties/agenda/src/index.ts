/**
 * @file Automatically generated by barrelsby.
 */

export * from "./AgendaModule.js";
export * from "./constants/constants.js";
export * from "./decorators/agenda.js";
export * from "./decorators/define.js";
export * from "./decorators/every.js";
export * from "./interfaces/AgendaStore.js";
export * from "./interfaces/interfaces.js";
export * from "./services/AgendaFactory.js";
