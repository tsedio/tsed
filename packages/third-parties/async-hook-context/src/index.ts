export * from "./decorators/injectContext";
export * from "./services/PlatformAsyncHookContext";
