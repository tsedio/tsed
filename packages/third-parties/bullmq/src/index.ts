/**
 * @file Automatically generated by barrelsby.
 */

export * from "./BullMQModule";
export * from "./config/config";
export * from "./constants/BullMQTypes";
export * from "./constants/constants";
export * from "./contracts/JobMethods";
export * from "./contracts/JobStore";
export * from "./decorators/InjectQueue";
export * from "./decorators/InjectWorker";
export * from "./decorators/JobController";
export * from "./dispatchers/JobDispatcher";
export * from "./dispatchers/JobDispatcherOptions";
export * from "./utils/createQueueProvider";
export * from "./utils/createWorkerProvider";
export * from "./utils/getJobToken";
export * from "./utils/getQueueToken";
export * from "./utils/getWorkerToken";
export * from "./utils/mapQueueOptions";
export * from "./utils/mapWorkerOptions";
