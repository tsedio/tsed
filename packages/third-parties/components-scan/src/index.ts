/**
 * @file Automatically generated by barrelsby.
 */

export * from "./cleanGlobPatterns.js";
export * from "./importFiles.js";
export * from "./importProviders.js";
export * from "./isTsEnv.js";
