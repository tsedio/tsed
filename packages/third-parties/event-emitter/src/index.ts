import {ConstructorOptions} from "eventemitter2";

declare global {
  namespace TsED {
    interface Configuration {
      eventEmitter?: {
        /**
         * Enable Event Emitter. Default `false`.
         */
        enabled?: boolean;
        /**
         * Disable event table displayed in the logger. Default `false`.
         */
        disableSummary?: boolean;
      } & ConstructorOptions;
    }
  }
}

export * from "./EventEmitterModule";
export * from "./services/EventEmitterFactory";
export * from "./decorators";
