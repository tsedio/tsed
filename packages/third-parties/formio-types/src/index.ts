/**
 * @file Automatically generated by barrelsby.
 */

export * from "./domain/FormioAccess.js";
export * from "./domain/FormioAction.js";
export * from "./domain/FormioActionInfo.js";
export * from "./domain/FormioActionItem.js";
export * from "./domain/FormioComponent.js";
export * from "./domain/FormioForm.js";
export * from "./domain/FormioPermission.js";
export * from "./domain/FormioRole.js";
export * from "./domain/FormioSchema.js";
export * from "./domain/FormioSubmission.js";
export * from "./domain/FormioToken.js";
