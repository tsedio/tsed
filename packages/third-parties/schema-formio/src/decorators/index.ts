export * from "./component";
export * from "./currency";
export * from "./dataSourceJson";
export * from "./dataSourceUrl";
export * from "./hidden";
export * from "./inputTags";
export * from "./multiple";
export * from "./password";
export * from "./select";
export * from "./tableView";
export * from "./textarea";
export * from "./errorMessage";
export * from "./errorLabel";
export * from "./minWords";
export * from "./maxWords";
export * from "./validate";
export * from "./form";
export * from "./tabs";
export * from "./placeholder";
export * from "./tooltip";
export * from "./label";
export * from "./customClass";
export * from "./modalEdit";
export * from "./mask";
export * from "./prefix";
export * from "./suffix";
export * from "./textCase";
export * from "./conditional";
export * from "./customConditional";
