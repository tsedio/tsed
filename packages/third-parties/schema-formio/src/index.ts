export * from "./components";
export * from "./decorators";
export * from "./registries/FormioMappersContainer";
export * from "./utils/getFormioSchema";
