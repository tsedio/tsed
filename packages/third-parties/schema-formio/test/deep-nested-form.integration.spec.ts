import {Default, Property} from "@tsed/schema";
import {Form, getFormioSchema, Label} from "../src";

@Form({label: "One hour before this maintenance"})
export class AutomationOneHourBefore {
  @Label("Remind subscribers")
  @Default(true)
  deliverNotifications: boolean = true;
}

@Form({label: "At the start time for this maintenance"})
export class AutomationAtStartTime {
  @Label("Set status to 'In Progress'")
  @Default(true)
  updateStatus: boolean = true;

  @Label("Set affected components' status to 'Under Maintenance'")
  @Default(true)
  setComponentStatus: boolean = true;

  @Label("Notify subscribers")
  @Default(true)
  deliverNotifications: boolean = true;
}

@Label("At the end of time for this maintenance")
export class AutomationAtEndTime {
  @Label("Set status to 'Completed'")
  @Default(true)
  updateStatus: boolean = true;

  @Label("Set affected components' status to 'Under Maintenance'")
  @Default(true)
  setComponentStatus: boolean = true;

  @Label("Notify subscribers")
  @Default(true)
  deliverNotifications: boolean = true;
}

@Label("Automation options test")
export class AutomationOptions {
  @Property()
  oneHourBefore: AutomationOneHourBefore;

  @Property()
  atStartTime: AutomationAtStartTime;

  @Property()
  atEndTime: AutomationAtEndTime;
}

@Form({
  included: {
    components: `/components`
  }
})
export class ScheduledMaintenance {
  @Property()
  automationOptions: AutomationOptions;
}

describe("Deep Nested form integration", () => {
  it("should generate form and nested form", async () => {
    const form = await getFormioSchema(ScheduledMaintenance);

    expect(form).toMatchSnapshot();
  });
});
