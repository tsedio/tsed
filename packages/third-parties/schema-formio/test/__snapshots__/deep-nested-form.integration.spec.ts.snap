// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deep Nested form integration should generate form and nested form 1`] = `
Object {
  "access": Array [],
  "components": Array [
    Object {
      "components": Array [
        Object {
          "collapsible": false,
          "components": Array [
            Object {
              "components": Array [
                Object {
                  "collapsible": false,
                  "components": Array [
                    Object {
                      "defaultValue": true,
                      "disabled": false,
                      "input": true,
                      "key": "deliverNotifications",
                      "label": "Remind subscribers",
                      "type": "checkbox",
                      "validate": Object {
                        "required": false,
                      },
                    },
                  ],
                  "disabled": false,
                  "input": false,
                  "key": "panel",
                  "label": "One hour before this maintenance",
                  "tableView": false,
                },
              ],
              "disabled": false,
              "display": "form",
              "input": true,
              "key": "oneHourBefore",
              "tableView": false,
              "type": "form",
              "validate": Object {
                "required": false,
              },
            },
            Object {
              "components": Array [
                Object {
                  "collapsible": false,
                  "components": Array [
                    Object {
                      "defaultValue": true,
                      "disabled": false,
                      "input": true,
                      "key": "updateStatus",
                      "label": "Set status to 'In Progress'",
                      "type": "checkbox",
                      "validate": Object {
                        "required": false,
                      },
                    },
                    Object {
                      "defaultValue": true,
                      "disabled": false,
                      "input": true,
                      "key": "setComponentStatus",
                      "label": "Set affected components' status to 'Under Maintenance'",
                      "type": "checkbox",
                      "validate": Object {
                        "required": false,
                      },
                    },
                    Object {
                      "defaultValue": true,
                      "disabled": false,
                      "input": true,
                      "key": "deliverNotifications",
                      "label": "Notify subscribers",
                      "type": "checkbox",
                      "validate": Object {
                        "required": false,
                      },
                    },
                  ],
                  "disabled": false,
                  "input": false,
                  "key": "panel",
                  "label": "At the start time for this maintenance",
                  "tableView": false,
                },
              ],
              "disabled": false,
              "display": "form",
              "input": true,
              "key": "atStartTime",
              "tableView": false,
              "type": "form",
              "validate": Object {
                "required": false,
              },
            },
            Object {
              "components": Array [
                Object {
                  "collapsible": false,
                  "components": Array [
                    Object {
                      "defaultValue": true,
                      "disabled": false,
                      "input": true,
                      "key": "updateStatus",
                      "label": "Set status to 'Completed'",
                      "type": "checkbox",
                      "validate": Object {
                        "required": false,
                      },
                    },
                    Object {
                      "defaultValue": true,
                      "disabled": false,
                      "input": true,
                      "key": "setComponentStatus",
                      "label": "Set affected components' status to 'Under Maintenance'",
                      "type": "checkbox",
                      "validate": Object {
                        "required": false,
                      },
                    },
                    Object {
                      "defaultValue": true,
                      "disabled": false,
                      "input": true,
                      "key": "deliverNotifications",
                      "label": "Notify subscribers",
                      "type": "checkbox",
                      "validate": Object {
                        "required": false,
                      },
                    },
                  ],
                  "disabled": false,
                  "input": false,
                  "key": "panel",
                  "label": "At the end of time for this maintenance",
                  "tableView": false,
                },
              ],
              "disabled": false,
              "display": "form",
              "input": true,
              "key": "atEndTime",
              "tableView": false,
              "type": "form",
              "validate": Object {
                "required": false,
              },
            },
          ],
          "disabled": false,
          "input": false,
          "key": "panel",
          "label": "Automation options test",
          "tableView": false,
        },
      ],
      "disabled": false,
      "display": "form",
      "input": true,
      "key": "automationOptions",
      "tableView": false,
      "type": "form",
      "validate": Object {
        "required": false,
      },
    },
  ],
  "display": "form",
  "included": Object {
    "components": "/components",
  },
  "machineName": "scheduled-maintenance",
  "name": "scheduled-maintenance",
  "submissionAccess": Array [],
  "tags": Array [],
  "title": "ScheduledMaintenance",
  "type": "form",
}
`;
